### PR TITLE
[charts/sn-platform]: Make Presto compatible for Istio mode

### LINKED ISSUE #825: [Private Cloud]: Make Presto compatible for Istio mode
- AuthorizationPolicy
- ServiceEntry
- DestinationRule

### PR DESCRIPTION
**Pull request by ciiiii**
Resolves #825 

### Motivation

Support Presto under Istio mode

### Modifications

- AuthorizationPolicy for coordinator/worker
- DestinationRules for coordinator/worker
- ServiceEntry for coordinator/worker

### Verifying this change

- [x] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Documentation

Check the box below.

Need to update docs? 

- [ ] `doc-required` 
  
  (If you need help on updating docs, create a doc issue)
  
- [x] `no-need-doc` 
  
  (Please explain why)
  
- [ ] `doc` 
  
  (If this PR contains doc changes)

### CODE REVIEW

**Review by fantapsody — APPROVED** [review]

File: charts/sn-platform/templates/presto/presto-serviceentry.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+{{- if and .Values.components.sql_worker .Values.istio.enabled }}
+{{- range $i, $e := until (int .Values.presto.coordinator.replicaCount ) }}
```
Comment (maxsxu) [inline charts/sn-platform/templates/presto/presto-serviceentry.yaml:21]: I think we can remove the unused `$e` variable here to simplify as below:

```
{{- range $i := until (int .Values.presto.coordinator.replicaCount ) }}
```

File: charts/sn-platform/templates/presto/presto-serviceentry.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    protocol: HTTPS
+  {{- else }}
+  - name: tcp-web
+    number: {{ $.Values.presto.coordinator.ports.http }}
+    protocol: HTTP
+  {{- end }}
+  resolution: DNS
+---
+{{- end }}
+{{- range $i, $e := until (int .Values.presto.worker.replicaCount) }}
```
Comment (maxsxu) [inline charts/sn-platform/templates/presto/presto-serviceentry.yaml:43]: I think we can remove the unused `$e` variable here to simplify as below:

```
{{- range $i := until (int .Values.presto.coordinator.replicaCount ) }}
```

Comment (ciiiii) [reply charts/sn-platform/templates/presto/presto-serviceentry.yaml]: applied, please review again

File: charts/sn-platform/templates/presto/presto-destinationrule.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+spec:
+  host: "{{ template "presto.hostname" . }}"
+  trafficPolicy:
+    tls:
+      mode: ISTIO_MUTUAL
+---
+apiVersion: networking.istio.io/v1beta1
+kind: DestinationRule
+metadata:
+  name: "{{ template "presto.coordinator" . }}-headless"
```
Comment (MarvinCai) [inline charts/sn-platform/templates/presto/presto-destinationrule.yaml:34]: I can't find headless service for service?

Comment (ciiiii) [reply charts/sn-platform/templates/presto/presto-destinationrule.yaml:34]: Pod dns records can be generated by Statefulset without headless Service, following the rule of `${podName}.${.spec.serviceName}.${namespace}.svc.cluster.local`.

ref: https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#stable-network-id

**Review by MarvinCai — APPROVED** [review]

**Pull request merged**